lib/mixpanel-node.js

```diff
@@ -576,6 +576,60 @@ var create_client = function(token, config) {
         },
 
         /**
+         people.union(distinct_id, data, callback)
+         ---
+         merge value(s) into a list-valued people analytics property.
+
+         usage:
+
+            mixpanel.people.union('bob', {'browsers': 'firefox'});
+
+            mixpanel.people.union('bob', {'browsers', ['chrome'], os: ['linux']});
+         */
+        union: function(distinct_id, data, callback) {
+            var $union = {};
+
+            if (typeof(data) !== 'object' || util.isArray(data)) {
+                if (metrics.config.debug) {
+                    console.error("Invalid value passed to mixpanel.people.union - data must be an object with array values");
+                }
+                return;
+            }
+
+            Object.keys(data).forEach(function(key) {
+                var val = data[key];
+                if (util.isArray(val)) {
+                    $union[key] = val;
+                } else if (typeof(val) !== 'object') {
```

Comment:
Everything looks great in this PR. The only issue I see is that this check is kind of overpermissive; it prevents sending objects as union vals but permits other types such as functions:
```js
mixpanel.people.union(distinct_id, {'woops': function() {}});
```
That's a contrived example, but some less obvious accidents will go through happily:
```js
mixpanel.people.union('bob', mixpanel);
// Sending the following data to Mixpanel (Engage):
// { '$union':
//    { token: [ 'mytoken' ],
//      send_request: [ [Function] ],
//      track: [ [Function] ],
//      import: [ [Function] ],
//      import_batch: [ [Function] ],
//      alias: [ [Function] ],
//      set_config: [ [Function] ] },
//   '$token': 'mytoken',
//   '$distinct_id': 'bob' }
```
Maybe just explicitly allowing only strings and numbers is the way forward?